definitions/checks/disk/available_space_var.rb

```diff
@@ -0,0 +1,26 @@
+module Checks
+  module Disk
+    class AvailableSpaceCandlepin < ForemanMaintain::Check
+      metadata do
+        label :available_space_cp
+        description 'Check to make sure /var/lib/candlepin has enough space'
+        tags :pre_upgrade
+      end
+
+      MAX_USAGE_IN_PERCENT = 90
+
+      def run
+        assert(enough_space?, "System has more than #{MAX_USAGE_IN_PERCENT}% space used"\
```

Comment:
Instead of assert can we give warning ? assert will fail the check and will need the whitelist option to bypass it. 
With that can we also work with candlepin team to fix the original issue, because hanging does not seem relevant on how much free disk space is available, i.e. 200 GB free space as you mentioned in issue?
If its happening on specific versions of foreman/katello then better we limit it in confine block until original issue is fixed ?

Reply:
I am OK with limit it to the affected versions. I experienced it with foreman-2.1.3, but I would assume it is the case for foreman >= 2.0 (or whenever candlepin started using/switched to Artemis).
But in this case it is IMHO valid to let the check fail, because it will break the system, as long as it is not fixed in Candlepin or more disk-space is assigned.

Reply:
Opened a bug-report for candlepin: https://bugzilla.redhat.com/show_bug.cgi?id=1898605

Reply:
Thank you for the changes, one last thing can we scope this for upstream only as of now ? Failing the check instead of warning concerns me for downstream, because this will error out in upgrade and will need to whitelist everytime even when having enough space ?

Reply:
I added `:warn => true` to `assert()`, is that enough?

Reply:
@upadhyeammit IMHO a warning makes zero sense here, since the whole reason for having this check is that the upgrade will fail if this check fails. (Even if you have lots of space, if it is less than 10% free, Candelpin/Artemis/whatever will refuse all tasks and the `foreman-maintain upgrade` run will fail.)

Reply:
Yes, I have thought over this and early failure makes sense, accordingly request to remove the `:warn =>true`.